integration/local-gadget/integration_test.go

```diff
@@ -0,0 +1,250 @@
+				Namespace: ns,
+			}
+
+			normalize := func(c *containercollection.Container) {
+				c.Name = ""
+				c.ID = ""
+				c.Pid = 0
+				c.OciConfig = nil
+				c.Bundle = ""
+				c.Mntns = 0
+				c.Netns = 0
+				c.CgroupPath = ""
+				c.CgroupID = 0
+				c.CgroupV1 = ""
+				c.CgroupV2 = ""
+				c.Labels = nil
+				c.PodUID = ""
+			}
+
+			var containers []*containercollection.Container
+			if err := json.Unmarshal([]byte(output), &containers); err != nil {
+				return err
+			}
+
+			for _, gotContainer := range containers {
+				normalize(gotContainer)
+				if reflect.DeepEqual(gotContainer, expectedContainer) {
+					return nil
+				}
+			}
+
+			return fmt.Errorf("output doesn't contain the expected container: %+v", expectedContainer)
+		},
+	}
+
+	commands := []*Command{
+		CreateTestNamespaceCommand(ns),
+		BusyboxPodRepeatCommand(ns, "sleep inf"),
+		WaitUntilTestPodReadyCommand(ns),
+		listContainersCmd,
+		DeleteTestNamespaceCommand(ns),
+	}
+
+	RunCommands(commands, t)
+}
+
+func TestListSingleContainer(t *testing.T) {
+	t.Parallel()
+	prefix := "test-list-single"
+	po := fmt.Sprintf("%s-pod", prefix)
+	cn := fmt.Sprintf("%s-container", prefix)
+	ns := GenerateTestNamespaceName(fmt.Sprintf("%s-namespace", prefix))
+
+	// TODO: Handle it once we support getting K8s container name for docker
+	// Issue: https://github.com/kinvolk/inspektor-gadget/issues/737
+	if *containerRuntime == ContainerRuntimeDocker {
+		t.Skip("Skip TestListSingleContainer on docker since we don't propagate the Kubernetes pod container name")
+	}
+
+	TestPodYaml := fmt.Sprintf(`
+apiVersion: v1
+kind: Pod
+metadata:
+  namespace: %s
+  name: %s
+spec:
+  restartPolicy: Never
+  terminationGracePeriodSeconds: 0
+  containers:
+  - name: %s
+    image: busybox
+    command: ["/bin/sh", "-c"]
+    args:
+    - "sleep inf"
+`, ns, po, cn)
+
+	listContainersCmd := &Command{
+		Name: "RunListSingleContainer",
+		Cmd:  fmt.Sprintf("local-gadget list-containers -o json --runtimes=%s --containername=%s", *containerRuntime, cn),
+		ExpectedOutputFn: func(output string) error {
+			expectedContainer := &containercollection.Container{
+				Name:      cn,
+				Podname:   po,
+				Runtime:   *containerRuntime,
+				Namespace: ns,
+			}
+
+			normalize := func(c *containercollection.Container) {
+				c.ID = ""
+				c.Pid = 0
+				c.OciConfig = nil
+				c.Bundle = ""
+				c.Mntns = 0
+				c.Netns = 0
+				c.CgroupPath = ""
+				c.CgroupID = 0
+				c.CgroupV1 = ""
+				c.CgroupV2 = ""
+				c.Labels = nil
+				c.PodUID = ""
+			}
+
+			var containers []*containercollection.Container
+			if err := json.Unmarshal([]byte(output), &containers); err != nil {
+				return err
+			}
+
+			for _, gotContainer := range containers {
+				normalize(gotContainer)
+				if !reflect.DeepEqual(gotContainer, expectedContainer) {
+					return fmt.Errorf("expect at leat one containers to match, expected=%+v, got=%+v", expectedContainer, gotContainer)
+				}
+			}
+
+			return nil
```

Comment:
This logic will make pass the test if the list of the returned containers is empty.